rust/main/hyperlane-base/src/settings/base.rs

```diff
@@ -145,11 +147,12 @@ macro_rules! build_contract_fns {
 type SequenceIndexer<T> = Arc<dyn SequenceAwareIndexer<T>>;
 
 impl Settings {
+    build_contract_fns!(build_application_operation_verifier, build_application_operation_verifiers -> dyn ApplicationOperationVerifier);
```

Comment:
I think creating operation verifiers here with `build_contract_fns` indicates that they interact with smart contracts. Wdyt about having a constructor that's more similar to `contract_syncs` instead? I'd ideally want to remove this `build_contract_fns` macro at some point

Reply:
`build_contract_fns` is already misused to instantiate `HyperlaneProvider` which does not depend on contract. We can plan to remove this macro as separate task.